src/cio_file_unix.c

```diff
@@ -112,33 +112,37 @@ int cio_file_native_map(struct cio_file *cf, size_t map_size)
 int cio_file_native_remap(struct cio_file *cf, size_t new_size)
 {
     int   result;
-    void *tmp;
+    void *tmp = NULL;
 
     result = 0;
 
+    if (new_size > cf->alloc_size) {
 /* OSX mman does not implement mremap or MREMAP_MAYMOVE. */
 #ifndef MREMAP_MAYMOVE
-    result = cio_file_native_unmap(cf);
+        result = munmap(cf->map, cf->alloc_size);
```

Comment:
isn't the code suggested the same as cio_file_native_unmap(cf); ?

Reply:
Indeed. I changed it in https://github.com/fluent/chunkio/pull/93/commits/1adcd3743b3d74438b2a3ead9cda1c4b17d9df15.